cse_helpers.py

```diff
@@ -1,69 +1,89 @@
+""" CSE Partial Factorization and Post-Processing
+
+The following script will perform partial factorization on SymPy expressions,
+which should occur before common subexpression elimination (CSE) to prevent the
+identification of undesirable patterns, and perform post-processing on the
+the resulting replaced/reduced expressions after the CSE procedure was applied.
+"""
+# Author: Ken Sible
+# Email:  ksible@outlook.com
+
 from SIMDExprTree import ExprTree
 import sympy as sp
 
-# Input:  sympyexpr = a single SymPy expression or list of SymPy expressions
-# Output: modified SymPy expression(s) where all integers and rationals are replaced
+# Input:  expr_list = single SymPy expression or list of SymPy expressions
+#         prefix    = string prefix for variable names (i.e. replacement symbols)
+#         ignore    = ignore factoring negative one (i.e. declare _NegativeOne_)
+#         factor    = perform partial factorization
+# Output: modified SymPy expression(s) where all integers and rationals were replaced
 #           with temporary placeholder variables that allow for partial factorization
 def cse_preprocess(expr_list, prefix='', ignore=False, factor=True):
+    if not isinstance(expr_list, list):
+        expr_list = [expr_list]
     def expand(a, n):
         if  n == 2:  return sp.Mul(a, a, evaluate=False)
         elif n > 2:  return sp.Mul(expand(a, n - 1), a, evaluate=False)
         return sp.Pow(expand(a, -n), -1, evaluate=False)
-    if not isinstance(expr_list, list):
-        expr_list = [expr_list]
-    var_map, sym_map = {}, {}
+    var_map, sym_map = {}, {} # Variable -> Rational, Rational -> Variable
```

Comment:
Let's name these variables to be consistent with the naming convention in SIMD.py.